transformations/suspecting_paraphraser/transformation.py

```diff
@@ -0,0 +1,203 @@
+from typing import List, Tuple
+
+from gender_extractor import GenderExtractor
+import nltk
+import numpy as np
+
+# Spacy needs this module, but it's used only implicitly
+import pyinflect  # noqa: F401
+import spacy
+
+from initialize import spacy_nlp
+from interfaces.QuestionAnswerOperation import QuestionAnswerOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+
+
+class SuspectingParaphraser(QuestionAnswerOperation):
+    """ This paraphraser transforms a yes/no question into a tag one.
+     
+    Example: "Did the American National Shipment company really break its own fleet?" 
+    -> "The American National Shipment company really broke its own fleet, didn't it?"
+    """
+    tasks = [TaskType.QUESTION_ANSWERING, TaskType.QUESTION_GENERATION]
+
+    languages = ["en"]
+
+    def __init__(self, seed=0, max_outputs=1, pronoun_mod=0.9):
+        super().__init__(seed, max_outputs=max_outputs)
+        np.random.seed(seed)
+        nltk.download("punkt")
+
+        self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")
+
+        self.gender_detector = GenderExtractor()
+        self.pronouns = ["he", "she", "it", "they"]
+        self.static_pronouns = ["i", "we", "you", *self.pronouns]
+
+        self._special_endings = {
+            "may": "may {} not",
+            "might": "might {} not",
+            "shall": "shan't {}",
+            "will": "won't {}",
+        }
+
+        self.pronoun_mod = pronoun_mod
+        self._pronoun_alt = (1 - pronoun_mod) / (len(self.pronouns) - 1)
+
+    def _transform(self, question):
+        text = nltk.word_tokenize(question)
+        doc = self.nlp(question)
+        modal = str(doc[0]).lower()
```

Comment:
Oh I missed this - why not loop through all docs instead of only the first one?

Reply:
It must be a leftover error, thank you for noticing this

Reply:
The [0] refers to the first token in the sentence, the name of the variable is a bit misleading here.

Reply:
Okay! Thanks for the confirmation!